command_line/rescale_diamond_anvil_cell.py

```diff
@@ -0,0 +1,302 @@
+cell, in which the sample is sandwiched between two anvils, effectively parallel flat
+plates of diamond.  The passage of the incident and diffracted beam through the
+anvils results in attenuation of both beams by the diamond by an amount that is
+dependent on the path length of each beam through each anvil.
+
+This utility calculates these path lengths and boosts the integrated reflection
+intensities to remove the calculated effect of the diamond attenuation.
+
+It is intended that this program be used to correct reflection intensities after
+integration but before scaling.  Call it on the output of dials.integrate.
+"""
+
+from __future__ import absolute_import, division, print_function
+
+import logging
+import os
+import sys
+
+import numpy as np
+from scipy.spatial.transform import Rotation
+
+from cctbx.eltbx import attenuation_coefficient
+from dials.array_family import flex
+import dials.util.log
+from dials.util.options import flatten_experiments, flatten_reflections, OptionParser
+from dxtbx.model import ExperimentList
+import libtbx.phil
+
+try:
+    from typing import List, Sequence, SupportsFloat
+except ImportError:
+    pass
+else:
+    Vector = Sequence[SupportsFloat]
+
+
+filename_root = os.path.splitext(os.path.basename(__file__))[0]
+logger = logging.getLogger("dials.%s" % filename_root)
+
+phil_scope = libtbx.phil.parse(
+    u"""
+    anvil
+        .caption = "Properties of the mounted diamond anvils"
+    {
+        density = 3510
+            .type = float
+            .help = "The density of the anvil material in kg per cubic metre.  "
+                    "The default is the typical density of diamond."
+
+        thickness = 1.5925
+            .type = float
+            .help = "The thickness in mm of each anvil in the pressure cell.\n"
+                    "The default is the thickness of the pressure cells in use on "
+                    "beamline I19 at Diamond Light Source."
+
+        normal = 0, 1, 0
+            .type = floats(size=3)
+            .help = "A 3-vector representing the normal to the anvil surfaces in the "
+                    "laboratory frame when the goniometer is at zero datum, i.e. the "
+                    "axes are all at zero degrees.  The vector may be given "
+                    "un-normalised."
+    }
+
+    output {
+        experiments = None
+            .type = path
+            .optional = True
+            .help = "The output experiment list file name.\n"
+                    "If None, don't output an experiment list file."
+        reflections = corrected.refl
+            .type = path
+            .help = "The output reflection table file."
+        log = dials.%s.log
+            .type = path
+    }
+    """
+    % filename_root
+)
+
+# Get the tabulated NIST mass attenuation coefficient data for carbon.
+carbon_attenuation_data = attenuation_coefficient.get_table("C")
+
+
+def correct_intensities_for_dac_attenuation(
+    elist,  # type: ExperimentList
```

Comment:
Is there any particular reason for using the variable names `elist` and `rtable` rather than `experiments` and `reflections` which appear to be much more commonly used throughout the codebase by a variety of authors? E.g.
https://github.com/dials/dials/blob/master/command_line/refine.py#L251
https://github.com/dials/dials/blob/master/command_line/index.py#L122
https://github.com/dials/dials/blob/master/command_line/scale.py#L151

Reply:
I was just trying to avoid shadowing the parameter names with the values passed to them in lines https://github.com/dials/dials/blob/f9862c6b068ca9fb58848a98f8d28bdf04f03a33/command_line/rescale_diamond_anvil_cell.py#L286-L288.  Old habits die hard, I guess.

Reply:
> For this pull request you shouldn't need a whole dataset added to dials-data - just the integrated experiments/reflections should be enough to exercise this command. To test `correct_intensities_for_dac_attenuation` you could probably even generate some fake experiments/reflections on the fly for the purposes of exercising the function.

OK, I'll grab you tomorrow to chat about it.  I'm not sure of the best way of measuring success/failure of such a test.